smokey/src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java

```diff
@@ -458,6 +475,17 @@ public void setRobotPosition(Pose2d updatedPosition)
     }
   }
 
+  /**
+   * A method to set the swerve drive mode
+   * @param swerveDriveMode
+   */
+  public void setSwerveDriveMode(SwerveDriveMode swerveDriveMode) {
+    this.swerveDriveMode = swerveDriveMode;
+    if (swerveDriveMode == SwerveDriveMode.IMMOVABLE_STANCE) {
+      this.chassisSpeeds = new ChassisSpeeds();
```

Comment:
I think accidental trigger press while actively moving could flip/tip the robot over as this will set wheel speeds to zero and break-ed.

Reply:
As in ... we need to tell John to use this with caution

Reply:
Yes. I am also concerned about this.  Trigger is very close to the bumpers, which he uses to up/down shift frequently while driving.

Should we move this command to a button that's more difficult to accidentally press, like D-pad or start/menu?

Reply:
Another idea is to only set the X pattern after the bot is already stopped.  Accidentally hitting the trigger would behave just as if he took his finger off the stick, with decel bringing us down to 0 gracefully.  

Implementation: when we enter IMMOVEABLE_STANCE mode, we do not immediately set the velocity to 0.  We let the bot decel to 0.0 as it would, and then only set the X pattern once the velocity has reached 0.0.

Reply:
implemented changing to immovable_stance only when the robot is not moving.